agent/hcp/telemetry/gauge_store.go

```diff
@@ -0,0 +1,75 @@
+package telemetry
+
+import (
+	"sync"
+
+	"go.opentelemetry.io/otel/attribute"
+)
+
+// Global store for Gauge values as workaround for async OpenTelemetry Gauge instrument.
+var once sync.Once
+var globalGauges *gaugeStore
+
+type gaugeStore struct {
+	store map[string]*gaugeValue
+	mutex sync.Mutex
```

Comment:
All my operations require a write, so I used `sync.Mutex` instead of RWMutex (since the store also deletes).

Reply:
Then I would say your "read" is also a write. Maybe extract the reading part and add a "delete" method?

Reply:
I never need to read only, as we need to always delete the value after reading. I'll also update the code to return early if the value doesn't exist. 

I feel like having it in a single method with a single write mutex makes it more atomic and simpler? By separating these, the callback will have to call read, then delete separately. Especially, as the gauge store won't be used anywhere else (I assume).

Is there a reason to extract both and prefer RWMutex?